### PR TITLE
executor: tiny refactor the `runtimeStats` collection code

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Better code

### What is changed and how it works?

Instead of calling `if runtimeStats != nil ...` check in every executor function,
we can just do it in the `executor.Next` function.
Remove redundant code and put the check in one place.

### Check List 

Tests 

 - No code

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @zz-jason @lysu

File: executor/executor.go [hunk 1]
```diff
@@ -182,14 +182,19 @@ type Executor interface {
...
+	base := e.base()
+	sessVars := base.ctx.GetSessionVars()
 	if atomic.CompareAndSwapUint32(&sessVars.Killed, 1, 0) {
 		return ErrQueryInterrupted
 	}
 	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan(fmt.Sprintf("%T.Next", e), opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
 	}
+	if base.runtimeStats != nil {
```
Comment (zz-jason) [inline executor/executor.go]: we should move it to the beginning of the function.

Comment (tiancaiamao) [reply executor/executor.go]: What's the difference?

Comment (zz-jason) [reply executor/executor.go]: We'd better cover the execution path as more as possible theoretically.

**Pull request merged**